### PR TITLE
feat: optimized flatten_fragments fn

### PR DESCRIPTION
**Pull request by mo-sameh**

### CODE REVIEW

**Comment by jalew188** [comment]
Hi @mo-sameh, nice PR. Unless it is a urgent fix, the PR must be always merged into `development` branch first, and all changes will be merged into `main`.

File: alphabase/peptide/fragment.py [hunk 1]
```diff
@@ -777,27 +864,19 @@ def remove_unused_fragments(
...
     
     Returns
     -------
     pd.DataFrame, List[pd.DataFrame]
         returns the reindexed precursor DataFrame and the sliced fragment DataFrames
     """
 
-    precursor_df = precursor_df.sort_values([frag_start_col], ascending=True)
-    frag_idx = precursor_df[[frag_start_col,frag_stop_col]].values
+    precursor_df = precursor_df.sort_values(['frag_start_idx'], ascending=True)
```
Comment (jalew188) [inline alphabase/peptide/fragment.py:874]: Put `frag_start_col` and `frag_stop_col` into paramters to make this fn also work for other kinds of input df

Comment (jalew188) [reply alphabase/peptide/fragment.py:874]: They are already parameters, why they are still constants here?

**Comment by GeorgWa** [comment]
Hi Feng,
The PR is just a draft at the moment. Mohamed and I discussed it yesterday. 
I will keep you in the loop next time.

**Comment by mo-sameh** [comment]
Hey Feng,

My bad, this was just to discuss some modifications with Georg. Gonna make
the changes and open a new one soon. Thanks for the heads-up!

**Comment by jalew188** [comment]
I see, no problem

**Comment by jalew188** [comment]
You can use draft merge for testing, see https://github.com/MannLabs/alpharaw/pull/23

**Pull request closed without merging**